azurerm/internal/services/network/application_gateway_resource.go

```diff
@@ -846,6 +846,12 @@ func resourceArmApplicationGateway() *schema.Resource {
 							Required: true,
 						},
 
+						"port": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validate.PortNumber,
+						},
```

Comment:
I've run the tests and just realised we're not reading this back in in the `flatten` function - can we update this to always export that e.g.:

```
port := 0
if props.Port != nil {
  port = int(*props.Port)
}
output["port"] = port
```

so the tests fail:

```
------- Stdout: -------
=== RUN   TestAccAzureRMApplicationGateway_probesWithPort
=== PAUSE TestAccAzureRMApplicationGateway_probesWithPort
=== CONT  TestAccAzureRMApplicationGateway_probesWithPort
TestAccAzureRMApplicationGateway_probesWithPort: testing.go:684: Step 0 error: Check failed: Check 2/2 error: azurerm_application_gateway.test: Attribute 'probe.0.port' expected "8082", got "0"
--- FAIL: TestAccAzureRMApplicationGateway_probesWithPort (763.77s)
FAIL
```

could we add this to the flatten function?

Reply:
Thanks, added this piece to `flattenApplicationGatewayProbes`.